### PR TITLE
Part3 - Constructors tasks

### PR DESCRIPTION
**Pull request by IntuitiveHarmony**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -83,26 +85,34 @@ struct Computer
...
 {
     return fileName != "" && clickSaveButton ? true : false;
 }
 
-bool Computer::runApp(bool doubleClick)
+void Computer::runApp(std::string text = "App is running")
 {
-    return doubleClick ? true : false;
+    std::cout << text << std::endl;
 }
```
Comment (matkatmusic) [inline main.cpp]: Default arguments should be defined where the function is declared, not where the function is defined. 
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -112,26 +122,40 @@ struct Car
...
+bool Car::stopMovement(bool applyBrakes = false)
 {
     return applyBrakes ? true : false;
 }
 
-bool Car::turn(bool rotateSteeringWheel)
+bool Car::turn(bool rotateSteeringWheel = true)
 {
     return rotateSteeringWheel ? true : false; 
 }
```
Comment (matkatmusic) [inline main.cpp]: Default arguments should be defined where the function is declared, not where the function is defined. 
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -205,29 +260,45 @@ struct PersonTwo
...
 
-int PersonTwo::jump(int crouch, int energy)
+int PersonTwo::jump(int crouch = 9, int energy = 4)
 {
     return crouch * energy;
 }
 
-bool PersonTwo::playPiano(bool memoryRecal, int nerves)
+bool PersonTwo::playPiano(bool memoryRecal = true, int nerves = 11)
```
Comment (matkatmusic) [inline main.cpp]: Default arguments should be defined where the function is declared, not where the function is defined. 
- [x] revise

File: main.cpp [hunk 4]
```diff
@@ -427,12 +551,94 @@ bool Bicycle::brake(int appliedPressure)
...
+    car.stopMovement();
+    car.turn();
+    car.honk();
+    std::cout << "Does it have 5 seats? " << (car.numberOfSeats >= 5 ? "Yes" : "No") << "\n\n";
+    
+    Tree tree;
+    tree.grow();
+    // This produceFruit is giving me issues again.  I do not understand the error it produces when I run it:
+        // in function `main':
+        // main.cpp:(.text+0x16e2): undefined reference to `Tree::produceFruit()'
```
Comment (matkatmusic) [inline main.cpp]: You **declared** a function.
You **called** that function somewhere.
You **did not define** that function. 
That is what **undefined** is referring to in this error message.

You need to start establishing the habit of searching for these types of error messages so you can figure out the solution yourself. 
https://www.google.com/search?q=cplusplus+undefined+reference+to&oq=cplus+plus+undeined&aqs=chrome.1.69i57j0i13i30j0i390l4.3734j0j7&sourceid=chrome&ie=UTF-8
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 4 above)
Comment (IntuitiveHarmony) [reply main.cpp]: My mistake was on line 199.  I was confused because I was defining here,  I just wasn't naming it properly

File: main.cpp [hunk 5]
```diff
@@ -140,27 +165,38 @@ struct Tree
...
+
+void Tree::grow()
 {
-    return sunLight * water;
+    std::cout << "All is well in the shade of the apple tree" << std::endl;
 }
-// Here it is a bit DRYer
-Tree::Fruit produceFruit(int grow)
+
+Tree::Fruit Tree::produceFruit(int grow = 7)
```
Comment (matkatmusic) [inline main.cpp]: define default param values where functions are declared, not where they are defined.
- [x] revise

File: main.cpp [hunk 6]
```diff
@@ -176,13 +212,46 @@ Tree::Fruit produceFruit(int grow)
     return fruit;
 }
 
-int Tree::photosynthesize(int sunLight, int water, int unkonwnFactor)
+int Tree::photosynthesize(int sunLight = 5, int water = 3, int unkonwnFactor = 7)
```
Comment (matkatmusic) [inline main.cpp]: - [x] revise

File: main.cpp [hunk 7]
```diff
@@ -176,13 +212,46 @@ Tree::Fruit produceFruit(int grow)
...
+    //     srand(time(0));
+    //     ~~~~~ ^~~~~~~
+    
+    // When I run this it give me the same number ever time I run the program.  I want it to give me a new random number each time:
+    // srand(5);
+    
+    srand(time(nullptr));
+    int low = 1;
+    int high = 10;
+    numberOfSeeds = (rand() % (high - low + 1)) + low;
```
Comment (matkatmusic) [inline main.cpp]: don't write code like this, per our slack discussion
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp [hunk 8]
```diff
@@ -222,6 +288,11 @@ bool PersonTwo::playPiano(bool memoryRecal, int nerves)
     return memoryRecal && nerves > 10 ? true : false; 
 }
 
+void PersonTwo::face()
```
Comment (matkatmusic) [inline main.cpp]: - [x] revise: function names should contain action words

File: main.cpp [hunk 9]
```diff
@@ -277,22 +345,30 @@ int Gears::shiftGear(int leftShifterValue, int rightShifterValue)
...
     float weight = 3.7f;
     char type = 'A';
     std::string material = "Carbon Fiber"; 
     std::string brand = "Acme";
     bool needMaintenance = false;
 
-    int adjustSeatHeight(int riderHeight);
+    void adjustSeatHeight(int riderHeight);
     bool shockAbsorbFrame(bool needMaintenance);
     bool supportWeight(bool needMaintenance);
```
Comment (matkatmusic) [inline main.cpp]: Don't use class names in the function names.
that's redundant
- [x] revise

File: main.cpp [hunk 10]
```diff
@@ -305,21 +381,30 @@ bool Frame::supportWeight(bool needMaintenanceFrame)
...
     float circumference = 30.2f;
     int numberOfSpokes = 25;
     std::string material = "Aluminum";
     std::string brand = "Acme";
-    bool needMaintenance = false;
+    bool needMaintenance = true;
 
     int spin(int speed, int distance); 
     bool shockAbsorbWheel(bool needMaintenance);
     bool supportWeightWheel(bool needMaintenance);
```
Comment (matkatmusic) [inline main.cpp]: Don't use class names in the function names.
that's redundant
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**